flarefly/fitter.py

```diff
@@ -157,39 +157,41 @@ def __init__(self, data_handler, name_signal_pdf, name_background_pdf, **kwargs)
             self._kde_signal_sample_ = []
             self._kde_signal_option_ = []
         else:
-            self._signal_pdf_ = [None for _ in enumerate(name_signal_pdf)]
-            self._hist_signal_sample_ = [None for _ in enumerate(name_signal_pdf)]
-            self._kde_signal_sample_ = [None for _ in enumerate(name_signal_pdf)]
-            self._kde_signal_option_ = [None for _ in enumerate(name_signal_pdf)]
+            self._signal_pdf_ = [None for _ in name_signal_pdf]
+            self._hist_signal_sample_ = [None for _ in name_signal_pdf]
+            self._kde_signal_sample_ = [None for _ in name_signal_pdf]
+            self._kde_signal_option_ = [None for _ in name_signal_pdf]
         if self._name_background_pdf_[0] == 'nobkg':
             self._background_pdf_ = []
             self._hist_bkg_sample_ = []
             self._kde_bkg_sample_ = []
             self._kde_bkg_option_ = []
         else:
-            self._background_pdf_ = [None for _ in enumerate(name_background_pdf)]
-            self._hist_bkg_sample_ = [None for _ in enumerate(name_background_pdf)]
-            self._kde_bkg_sample_ = [None for _ in enumerate(name_background_pdf)]
-            self._kde_bkg_option_ = [None for _ in enumerate(name_background_pdf)]
-        self._name_refl_pdf_ = kwargs.get('name_refl_pdf', [None for _ in enumerate(name_signal_pdf)])
+            self._background_pdf_ = [None for _ in name_background_pdf]
+            self._hist_bkg_sample_ = [None for _ in name_background_pdf]
+            self._kde_bkg_sample_ = [None for _ in name_background_pdf]
+            self._kde_bkg_option_ = [None for _ in name_background_pdf]
+        self._name_refl_pdf_ = kwargs.get('name_refl_pdf', [None for _ in name_signal_pdf])
         if len(self._name_refl_pdf_) != len(self._name_signal_pdf_):
             Logger('List of pdfs for signals and reflections different! Exit', 'FATAL')
-        self._refl_pdf_ = [None for _ in enumerate(name_signal_pdf)]
-        self._hist_refl_sample_ = [None for _ in enumerate(name_signal_pdf)]
-        self._kde_refl_sample_ = [None for _ in enumerate(name_signal_pdf)]
-        self._kde_refl_option_ = [None for _ in enumerate(name_signal_pdf)]
-        self._refl_over_sgn_ = [0. for _ in enumerate(name_signal_pdf)]
+        if self._name_signal_pdf_[0] == 'nosignal':
```

Comment:
There is already such a condition a few lines above I belive. Could you put the line 178 in this block?

Reply:
Hi @AlexBigO, thanks a lot for your comment!
I have implemented it

Reply:
Thanks!

PS: I guess my comment was posted twice because I got some connection issues when posting the comment :sweat_smile: